standards/request-response.md

```diff
@@ -444,8 +444,32 @@ Access-Control-Allow-Methods: *
 
 <hr />
 
-```note
-**KEEP AN EYE ON IT**: The [Idempotency-Key Request header](https://tools.ietf.org/id/draft-idempotency-header-01.html) is in an experimental state but getting lots of attention as a pattern of making fault-tolerant resilient requests for traditionally non-idempotent methods like `POST`.
+#### [Idempotency-Key Request header](https://datatracker.ietf.org/doc/html/draft-idempotency-header-01)
+**Type**: Request
+
+**Support**: OPTIONAL
+
+**Description**:
+The HTTP Idempotency-Key request header field can be used to carry an "idempotency key" to make non-idempotent HTTP methods such as POST or PATCH fault-tolerant.
+This is an experimental header, as there is common usage of it, but no accepted spec for it.
+`Idempotency-Key` should typically be a V4 UUID as a string, or another random string with enough entropy to avoid collisions, and 
+should be no longer than 255 characters long.
+An API must in some way persist the Idempotency-Key in a stateful way to manage the idempotency.
+It is common to include the key as part of the primary key or as a secondary index for the resource.
+
+- The header value **MUST** follow this regex format: `[a-z0-9-]{20,255}`
+- An API **MUST** return a `400` when the header value is an invalid format.
```

Comment:
> Is there a restriction in the types of Verbs you should use this header with, such as POST, PATCH (PUT?DELETE?)

Not sure about restrictions. The draft spec mentioned PATCH as another possibility, but I honestly don't know of a use case to use the header outside of a POST.  You can use an idempotency-key for any request that could possibly not be idempotent, and it might be hard to imagine all of the use cases or want to lock it down if there is some weird reason you might want to use it for a DELETE or something.  I don't think there's any harm to being vague about it's usage outside of it being a request header.

> Is this considered sensitive data that should not be logged?
It should, in general, not be sensitive data.  And it would be OK to log.

> Any consideration around the expiration of idempotency keys at all

I think it would be reasonable for some APIs to have short lived ones, and reasonable for some APIs to have indefinate ones. It seems reasonable that an API might want to just have a short lived one for a particular request, and once the request is fully processed, not really need to save the key anymore, and you can save some space and get index speed efficiency that way.
payload-storage-service, on the other hand, will bind a key to a payload for the entire retention policy of that payload, since it gets baked into the ID of the payload.

> Any scope considerations... is an idempotency key scoped to a resource? to an endpoint? other?

That might be a bit hard to say.  In a pure definitional sense it's an ID that identifies a particular unique "request", including retries, almost like a request session of a sort.  Perhaps a better way to think about it is that it works a bit more like a way to identify a unique "action", which you could imagine an action having multiple requests (if anything, a single endpoint with multiple retries).  I think almost always it will just be a way to have a unique POST behavior.